wireapp/wire-android
# fix: App crash on app unlock with fingerprint (WPB-5110) (WPB-5108)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5110" title="WPB-5110" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-5110</a>  [Android] Applock - Fail applock biometrics more than 3 times , Crash
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

1. App crash when unlock with biometrics fails due to too many failed attempts
2. Black screen shown without showing biometrics prompt dialog

### Causes (Optional)

1. After 5 failed attempts of auth with biometrics, the system will lock out the biometrics API and will return an error which we were not handling it.

2. The app was emitting the screen to go when it's in the background causing a failure when showing biometrics prompt.
`Unable to start authentication. Called after onSaveInstanceState()`


### Solutions

1. handle that error by navigating the user to passcode screen

2. Make flow observation lifecycle aware so the app will stop emitting/observing when it's in background. Also this way the app will prevent resource wasting.

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

1. 
- Try to authenticate with wrong biometrics multiple time so the API get locked out.
- Open the app again
- The app should not crash

2.
- Authenticate with biometrics
- Close the app and let it in background for 1min
- Open the app again, you should see biometrics dialog

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
